File: browser-use-node/examples/structured-output.ts
/**
 * Structured output — get typed results using Zod schemas.
 */
import "dotenv/config";
import { BrowserUse } from "browser-use-sdk";
import { z } from "zod";

async function main() {
  const client = new BrowserUse();

  const HackerNewsResult = z.object({
    posts: z.array(
      z.object({
        title: z.string(),
        url: z.string(),
        score: z.number(),
      }),
    ),
  });

  const result = await client.run(
    "Find the top 10 Hacker News posts. Return title, url, and score for each.",
    { schema: HackerNewsResult },
  );

  if (result.output) {
    for (const post of result.output.posts) {
      console.log(`${post.score} - ${post.title}`);
      console.log(`  ${post.url}\n`);
    }
  }
}

main();

File: browser-use-python/examples/structured_output.py
"""
Structured output -- get typed results using Pydantic models.

Pass output_schema to run() and the SDK handles JSON schema conversion
and response parsing automatically.
"""

import asyncio

from dotenv import load_dotenv
from pydantic import BaseModel
from browser_use_sdk import AsyncBrowserUse

load_dotenv()


class HackerNewsPost(BaseModel):
    title: str
    url: str
    score: int


class SearchResult(BaseModel):
    posts: list[HackerNewsPost]


async def main():
    client = AsyncBrowserUse()

    # await run() with output_schema returns a fully typed Pydantic model
    parsed = await client.run(
        "Find the top 10 Hacker News posts. Return title, url, and score for each.",
        output_schema=SearchResult,
    )

    if parsed is not None:
        for post in parsed.posts:
            print(f"{post.score} - {post.title}")
            print(f"  {post.url}\n")


asyncio.run(main())
